File: atividade05/python/main.py
horaAdmin = 22.00
horaProd = 12.00
taxaInss = 0.07
taxaIr = 0.12
planoDeSaude = 20.00
MaxFuncionarios = 5

funcionarios = []

def calcularSalarioBase(departamento, horasTrabalhadas):
    valorHora = horaAdmin if departamento == 1 else horaProd
    return horasTrabalhadas * valorHora

def calcularHoraExtra(departamento, horasTrabalhadas):
    if horasTrabalhadas <= 40:
        return 0.0
    valor_hora = horaAdmin if departamento == 1 else horaProd
    return valor_hora * 2 * (horasTrabalhadas - 40)

def calcularInsalubridade(departamento, salarioBase):
    return salarioBase * 0.15 if departamento == 2 else 0.0

def calcularbonificacao(departamento, horasTrabalhadas, salarioBase):
    if departamento != 1:
        return 0.0
    if horasTrabalhadas > 40:
        return salarioBase * 0.10
    elif horasTrabalhadas > 30:
        return salarioBase * 0.05
    elif horasTrabalhadas > 20:
        return salarioBase * 0.03
    return 0.0

def calcularSalarioBruto(salarioBase, horaExtra, insalubridade, bonificacao):
    return salarioBase + horaExtra + insalubridade + bonificacao

def calcularInss(salarioBruto):
    return salarioBruto * taxaInss

def calcularImpostoDeRenda(salarioBruto):
    return salarioBruto * taxaIr

def calcularSalarioLiquido(salarioBruto, inss, impostoDeRenda):
    return salarioBruto - inss - impostoDeRenda

def exibirFolhaDePagamento(funcionarios):
    cabecalho = (
        "Ordem    Nome           Salário Base   Hora Extra  Insalubridade   Bonificação  INSS      IR        Plano Saúde  Salário Líquido\n"
        "----------------------------------------------------------------------------------------------------------------------------"
    )
    print(cabecalho)

    for i, func in enumerate(funcionarios):
        linha = (
            f"{i + 1:<5}  {func['nome']:<15} {func['salario_base']:<15.2f} {func['hora_extra']:<10.2f} "
            f"{func['insalubridade']:<15.2f} {func['bonificacao']:<12.2f} {func['inss']:<10.2f} "
            f"{func['ir']:<10.2f} {planoDeSaude:<15.2f} {func['salario_liquido']:<15.2f}"
        )
        print(linha)

for i in range(MaxFuncionarios):
    print(f"Funcionario {i + 1}")
    nome = input("Nome: ")
    departamento = int(input("Departamento (Digite 1 para Administração ou 2 para Produção)"))
    horasTrabalhadas = float(input("Horas trabalhadas: "))

    salarioBase = calcularSalarioBase(departamento, horasTrabalhadas)
    horaExtra = calcularHoraExtra(departamento, horasTrabalhadas)
    insalubridade = calcularInsalubridade(departamento, salarioBase)
    bonificacao = calcularbonificacao(departamento, horasTrabalhadas, salarioBase)
    salarioBruto = calcularSalarioBruto(salarioBase, horaExtra, insalubridade, bonificacao)
    inss = calcularInss(salarioBruto)
    impostoDeRenda = calcularImpostoDeRenda(salarioBruto)
    salarioLiquido = calcularSalarioLiquido(salarioBruto, inss, impostoDeRenda)

    funcionario = {
        'nome': nome,
        'departamento': departamento,
        'horas_trabalhadas': horasTrabalhadas,
        'salario_base': salarioBase,
        'hora_extra': horaExtra,
        'insalubridade': insalubridade,
        'bonificacao': bonificacao,
        'salario_bruto': salarioBruto,
        'inss': inss,
        'ir': impostoDeRenda,
        'salario_liquido': salarioLiquido
    }
    funcionarios.append(funcionario)

exibirFolhaDePagamento(funcionarios)

File: atividade05/cpp/main.cpp
#include <iostream>
#include <iomanip>
#include <string>

using namespace std;

const double horaAdmin = 22.00;
const double horaProducao = 12.00;
const double taxaInss = 0.07;
const double taxaIr = 0.12;
const double planoDeSaude = 20.00;
const int NumMaxFuncionario = 5;

struct Funcionario {  //classe do funcionário
    string nome;
    int departamento;
    double horasTrabalhadas;
    double salarioHoras;
    double horasExtra;
    double insalubridade;
    double bonificacao;
    double salarioBruto;
    double inss;
    double impostoDeRenda;
    double salarioLiquido;
};

double calcularSalarioBase(int departamento, double horasTrabalhadas) {
    double valorHora = (departamento == 1) ? horaAdmin : horaProducao;
    return horasTrabalhadas * valorHora;
}

double calcularHoraExtra(int departamento, double horasTrabalhadas) {
    if (horasTrabalhadas <= 40) return 0.0; //fez ou não hora extra
    double valorHora = (departamento == 1) ? horaAdmin : horaProducao; //if else pra prod ou admin e passa o valor abaixo
    return  valorHora * 2 * (horasTrabalhadas - 40);
}

double calcularInsalubridade(int departamento, double salarioHoras) {
    return (departamento == 2 ) ? salarioHoras * 0.15 : 0.0;
}

double calcularBonificacao(int departamento, double horasTrabalhadas, double salarioHoras) {
    if (departamento != 1) return 0.0;
    if (horasTrabalhadas > 40) return salarioHoras * 0.10;
    if (horasTrabalhadas > 30) return salarioHoras * 0.05;
    if (horasTrabalhadas > 20) return salarioHoras * 0.03;
    return 0.0;
}

double calcularSalarioBruto(double salarioHoras, double horasExtra, double insalubridade, double bonificacao) {
    return salarioHoras + horasExtra + insalubridade + bonificacao;
}

double calcularInss(double salarioBruto) {
    return salarioBruto * taxaInss;
}

double calcularImpostoDeRenda(double salarioBruto) {
    return salarioBruto * taxaIr;
}

double calcularSalarioLiquido(double salarioBruto, double inss, double impostoDeRenda){
    return salarioBruto - impostoDeRenda - inss - planoDeSaude;
}

void exibirFolhaDePagamento(const Funcionario funcionarios[], int tamanhoMax) {
    cout << "Ordem  Nome           Salario Base   Hora Extra  Insalubridade   Bonificacao  INSS      IR        Plano Saude  Salario Liquido\n";
    cout << "----------------------------------------------------------------------------------------------------------------------------\n";

    for (int i = 0; i < tamanhoMax; ++i) {
        const Funcionario& func = funcionarios[i];
        cout << setw(5) << i + 1 << "  " //width
             << setw(15) << left << func.nome << " "
             << setw(15) << fixed << setprecision(2) << func.salarioHoras << " "
             << setw(10) << fixed << setprecision(2) << func.horasExtra << " "
             << setw(15) << fixed << setprecision(2) << func.insalubridade << " "
             << setw(12) << fixed << setprecision(2) << func.bonificacao << " "
             << setw(10) << fixed << setprecision(2) << func.inss << " "
             << setw(10) << fixed << setprecision(2) << func.impostoDeRenda << " "
             << setw(15) << fixed << setprecision(2) << planoDeSaude << " "
             << setw(15) << fixed << setprecision(2) << func.salarioLiquido << "\n";
    }
}

int main(){

    Funcionario funcionarios[NumMaxFuncionario];

    for (int i = 0; i < NumMaxFuncionario; ++i) {
        cout << "Funcionario " << i + 1 << ":\n";
        cout << "Nome: ";
        cin >> funcionarios[i].nome;
        cout << "Departamento (Digite 1 para Administrativo ou 2 para Producao): ";
        cin >> funcionarios[i].departamento;
        cout << "Horas trabalhadas: ";
        cin >> funcionarios[i].horasTrabalhadas;

        funcionarios[i].salarioHoras = calcularSalarioBase(funcionarios[i].departamento, funcionarios[i].horasTrabalhadas);
        funcionarios[i].horasExtra = calcularHoraExtra(funcionarios[i].departamento, funcionarios[i].horasTrabalhadas);
        funcionarios[i].insalubridade = calcularInsalubridade(funcionarios[i].departamento, funcionarios[i].salarioHoras);
        funcionarios[i].bonificacao = calcularBonificacao(funcionarios[i].departamento, funcionarios[i].horasTrabalhadas, funcionarios[i].salarioHoras);
        funcionarios[i].salarioBruto = calcularSalarioBruto(funcionarios[i].salarioHoras, funcionarios[i].horasExtra, funcionarios[i].insalubridade, funcionarios[i].bonificacao);
        funcionarios[i].inss = calcularInss(funcionarios[i].salarioBruto);
        funcionarios[i].impostoDeRenda = calcularImpostoDeRenda(funcionarios[i].salarioBruto);
        funcionarios[i].salarioLiquido = calcularSalarioLiquido(funcionarios[i].salarioBruto, funcionarios[i].inss, funcionarios[i].impostoDeRenda);
    }

    exibirFolhaDePagamento(funcionarios, NumMaxFuncionario);

    return 0;
}
